UPGRADE.md

```diff
@@ -0,0 +1,251 @@
+`phpDocumentor` implementation details, but is now fully independent and
+only relying on `string` type definitions. Therefore:
+
+ * `ReflectionType::getTypeObject()` was removed
+ * `ReflectionType::createFromType()` now requires a `string` for the
+   `$type` parameter
+
+### `BetterReflection\Reflection\ReflectionFunctionAbstract#setReturnType()` now requires a `string`
+
+The `BetterReflection\Reflection\ReflectionFunctionAbstract#setReturnType()` method
+now requires a `string` argument to be passed to it. The type will be
+detected from the given string.
+
+### `Roave\BetterReflection\SourceLocator\Reflection\SourceStubber` now works with core `ReflectionClass`
+
+The `Roave\BetterReflection\SourceLocator\Reflection\SourceStubber` is 
+now capable of working with just `ReflectionClass` instances from PHP
+core: it is no longer required to use `Zend\Code` instances, but the
+type hints also changed accordingly.
+
+### `Roave\BetterReflection\Reflector\ClassReflector::buildDefaultReflector()` dropped
+
+Since the `Roave\BetterReflection\BetterReflection` kernel was introduced,
+`Roave\BetterReflection\Reflector\ClassReflector::buildDefaultReflector()` was
+removed. The equivalent API is following:
+
+```php
+<?php
+
+use Roave\BetterReflection\BetterReflection;
+
+$reflector = (new BetterReflection())->classReflector();
+```
+
+### `Roave\BetterReflection\Reflection\Exception\PropertyNotPublic` dropped
+
+The `Roave\BetterReflection\Reflection\Exception\PropertyNotPublic` was
+thrown when a non-accessible reflection property is being read or written
+to. Since this is "better" reflection, there is no need for calling anything
+like `setAccessible` on a `Roave\BetterReflection\Reflection\ReflectionProperty`
+instance. All properties are directly accessible, even if `private`, `protected`
+or dynamically defined.
+
+### `Roave\BetterReflection\Reflector\FunctionReflector` constructor changed
+
+The constructor of `Roave\BetterReflection\Reflector\FunctionReflector`
+changed in its required parameters. For generic `BetterReflection` usage,
+it is advisable to simply obtain a `Roave\BetterReflection\Reflector\FunctionReflector`
+from the `Roave\BetterReflection\BetterReflection` kernel:
+
+```php
+<?php
+
+use Roave\BetterReflection\BetterReflection;
+
+$reflector = (new BetterReflection())->functionReflector();
+```
+
+### `Roave\BetterReflection\SourceLocator\Ast\Locator` constructor changed
+
+The constructor of `Roave\BetterReflection\SourceLocator\Ast\Locator`
+changed in its required parameters. For generic `BetterReflection` usage,
+it is advisable to simply obtain a `Roave\BetterReflection\SourceLocator\Ast\Locator`
+from the `Roave\BetterReflection\BetterReflection` kernel:
+
+```php
+<?php
+
+use Roave\BetterReflection\BetterReflection;
+$astLocator = (new BetterReflection())->astLocator();
+```
+
+### `Roave\BetterReflection\Reflection\Reflection*::createFromNode()` changes
```

Comment:
Should we mark all these methods as `@internal`?

Reply:
@asgrim possibly a good idea

Reply:
see #358